### PR TITLE
chore: Let OTPInput reset value on demand by parent

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR improve the handling on OTPInput value change to help the handling of the value by the parent component.
Adds accessibilityLabel as prop to improve accessibility

## List of changes proposed in this pull request
- Handle the value change only by parent
- Add accessibilityLabel as component prop

## How to test
Check OTPInput component on example app page

### CODE REVIEW

File: example/src/pages/OTPInput.tsx [hunk 1]
```diff
@@ -30,14 +31,19 @@ const OTPWrapper = ({ secret = false, validation = false }: WrapperProps) => {
...
-      value={value}
-      onValueChange={onValueChange}
-      length={OTP_LENGTH}
-      secret={secret}
-      onValidate={onValidate}
-      errorMessage={"Wrong OTP"}
-    />
+    <>
+      <OTPInput
+        value={value}
```
Comment (shadowsheep1) [inline example/src/pages/OTPInput.tsx:36]: What about exposing also a view ref? As for a11y this'd let us focus on the element if we want. What do you think? The alternative is to wrap the componenti in an accessible `View` with a ref.

```ts
 <View accessible ref={pinPadViewRef}>
  <OTPInput
    secret
    value={pin}
    accessibilityLabel={I18n.t(
      "authentication.cie.pin.accessibility.label"
    )}
    onValueChange={setPin}
    length={CIE_PIN_LENGTH}
  />
</View>
```

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: example/src/pages/OTPInput.tsx (hunk 1 above)
Comment (CrisTofani) [reply example/src/pages/OTPInput.tsx:36]: Added in [1a92ac5](https://github.com/pagopa/io-app-design-system/pull/248/commits/1a92ac5c175733a233550484d3ad081e534f409f)

File: example/src/pages/OTPInput.tsx [hunk 2]
```diff
@@ -30,14 +31,19 @@ const OTPWrapper = ({ secret = false, validation = false }: WrapperProps) => {
...
+        accessibilityLabel={"OTP Input"}
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply example/src/pages/OTPInput.tsx:37]: Added in [1a92ac5](https://github.com/pagopa/io-app-design-system/pull/248/commits/1a92ac5c175733a233550484d3ad081e534f409f)

File: src/components/otpInput/OTPInput.tsx [hunk 3]
```diff
@@ -31,105 +33,112 @@ type Props = {
...
-                ? "focus"
-                : "default"
-            }
-            secret={secret}
-            value={inputValue[i]}
+          accessibilityLabel={accessibilityLabel}
+          accessibilityHint={accessibilityHint}
+          accessibilityValue={{ text: value }}
+        >
+          <TextInput
```
Comment (shadowsheep1) [inline src/components/otpInput/OTPInput.tsx:106]: I've found that we are not exposing `InputAccessoryViewID` prop too? We actually use it in app. What do you think?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: src/components/otpInput/OTPInput.tsx (hunk 3 above)
Comment (CrisTofani) [reply src/components/otpInput/OTPInput.tsx:108]: addressed [9f8d73b](https://github.com/pagopa/io-app-design-system/pull/248/commits/9f8d73b3ac64da3c014906893c88d361f4e95925)

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**